MIG 885 storageclass_conversion (#42719)
openshift/openshift-docs: modules/migration-converting-the-storage-class-of-persistent-volumes-cam.adoc

// Module included in the following assemblies:
//
// * migration_toolkit_for_containers/advanced-migration-options-mtc.adoc

:_content-type: PROCEDURE
[id="migration-converting-the-storage-class-of-persistent-volumes-cam_{context}"]
= Converting the storage class of persistent volumes in the {mtc-short} web console

> You can convert the storage class of persistent volumes (PVs) by migrating them within the same cluster. For that, you must create and run a migration plan in the {mtc-full} ({mtc-short}) web console.

Reviewer: ```suggestion
You can convert the storage class of persistent volumes (PVs) by migrating them within the same cluster. To do so, you must create and run a migration plan in the {mtc-full} ({mtc-short}) web console.
```

Author: Done

.Prerequisites

* You must be logged in as a user with `cluster-admin` privileges on the cluster on which {mtc-short} is running.
* You must add the cluster to the {mtc-short} web console.

.Procedure

> . In the left-side navigation pane of the web console, click *Projects*.

Author: OCP added

. In the list of projects, click your project.
+
The *Project details* page opens.

> . Click the DeploymentConfig. Note the name of its running pod.

Reviewer: > Click the DeploymentConfig.

Suggestion: Click the DeploymentConfig button.

Avoid using UI elements as words.

Author: Changed

. Open the YAML tab of the project. Find the PVs and note the names of their corresponding persistent volume claims (PVCs).
. In the {mtc-short} web console, click *Migration plans*.
. Click *Add migration plan*.
. Enter the *Plan name*.
+

> The migration plan name must not exceed 253 lower-case alphanumeric characters (`a-z, 0-9`) and must not contain spaces or underscores (`_`).

Author: Done

. In *Migration type*, select *Storage class conversion*.

> . In *Source cluster*, select *host*.

Reviewer 2: In *Source cluster* , select the desired cluster for storage class conversion.

Author: @stillalearner  Mike Turley wrote this: 
"...select "host" as our source cluster (the "host" cluster is automatically configured when MTC is installed, it is the cluster MTC is running on)"
Is it wrong?

Reviewer 2: @sbeskin-redhat  the storageclass conversion can be performed on any of the clusters (source or target) added in the UI, not necessary the host only.

Author: Changed

. Click *Next*.
+
The *Namespaces* page opens.
. Select the required project.
. Click *Next*.
+
The *Persistent volumes* page opens. The page displays the PVs in the project, all selected by default.

> . For each PV, select the desired target storage class.

Reviewer 2: target storage class is a field name.

Author: @stillalearner  I mean one of the possible menu options here, not the name of the field.

. Click *Next*.
+
The wizard validates the new migration plan and shows that it is ready.
. Click *Close*.
+
The new plan appears on the *Migration plans* page.

> . To start the conversion, click the three dots menu of the new plan and select *Migrations*, then *Cutover*.

Reviewer 2: To start the conversion, click the three dots menu of the new plan and select *Cutover* option visible under the Migrations.

Author: Changed to "select *Migrations* -> *Cutover*"

> . To start the conversion, click the three dots menu of the new plan and select *Migrations*, then *Cutover*.

Reviewer 2: So, actually there are 2 options

Stage & Cutover

Shall we mention the description of both and ask the user to choose appropriately?

Author: @stillalearner Is the Stage option relevant for storageclass conversion?

Reviewer 2: @sbeskin-redhat  yes, it is. Here are the details of both migrations. Stage and Cutover

During a stage migration:
PV data is copied to the converted PVs.
PVC references in the applications are **not** updated to new PVCs.

During a cutover migration:
All transactions on source applications are halted for the duration of the migration.
PVC references in the applications are updated to new PVCs before restarting the applications.

Author: @stillalearner  So what are the considerations for choosing one of those options? And if the user chooses Stage, how does the migration proceeds?

+
The *Cutover migration* notification appears.
. Click *Migrate*.
+
The *Cutover started* message appears.

> .  To see the status of the current migration, click the number in the *Migration* column.

Reviewer 2: ** Migrations ** column

Author: Corrected

+
The *Migrations* page opens.

> . To see more details on the current migration and monitor its progress, click the *Type* column.

Reviewer 2: To see more details on the current migration and monitor its progress, choose the migration from the *Type* column.

Reason for suggested change: There can be multiple migrations for a single migplan.

Author: Done

+
The *Migration details* page opens.
When the migration progresses to the DirectVolume step and the status of the step becomes `Running Rsync Pods to migrate Persistent Volume data`, you can click *View details* and see the detailed status of the copies.

> . Open the *PersistentVolumeClaims* tab.

Reviewer 2: There is no PersistentVolumeClaims tab in mtc console, is here also ocp console need to be mentioned ?

Reviewer 2: Its better to keep both the ocp console steps to check PVC and Pods at the end after completion of migration.

Author: @stillalearner  Yes, I've added OCP console.
Agreed, changed

// Module included in the following assemblies:
//
// * migration_toolkit_for_containers/advanced-migration-options-mtc.adoc

:_content-type: PROCEDURE
[id="migration-converting-the-storage-class-of-persistent-volumes-cam_{context}"]

> = Converting the storage class of persistent volumes in the {mtc-short} web console

Reviewer 3: Suggest changing title to "Converting storage classes".

If a CLI procedure exists, it should be documented.

> You can convert the storage class of persistent volumes (PVs) by migrating them within the same cluster. To do so, you must create and run a migration plan in the {mtc-full} ({mtc-short}) web console.

Reviewer 3: Either "storage classes of persistent volumes" or "the storage class of a persistent volume"

. In the list of projects, click your project.
+
The *Project details* page opens.
. Click the *DeploymentConfig* name. Note the name of its running pod.
. Open the YAML tab of the project. Find the PVs and note the names of their corresponding persistent volume claims (PVCs).
. In the {mtc-short} web console, click *Migration plans*.
. Click *Add migration plan*.
. Enter the *Plan name*.
+
The migration plan name must contain 3 to 63 lower-case alphanumeric characters (`a-z, 0-9`) and must not contain spaces or underscores (`_`).

. In *Migration type*, select *Storage class conversion*.
. In *Source cluster*, select the desired cluster for storage class conversion.
. Click *Next*.
+
The *Namespaces* page opens.
. Select the required project.
. Click *Next*.
+
The *Persistent volumes* page opens. The page displays the PVs in the project, all selected by default.
. For each PV, select the desired target storage class.
. Click *Next*.
+
The wizard validates the new migration plan and shows that it is ready.
. Click *Close*.
+
The new plan appears on the *Migration plans* page.
. To start the conversion, click the three dots menu of the new plan and select *Migrations*.

> +

Reviewer 2: In the three dots menu, "Migrations" is not a selectable object, It is just a heading (plain text) which is always there and under which the Stage and Cutover are the buttons.

Author: Corrected

Author: @stillalearner 
Sachin, please approve this PR. Thank you!
